File: android/app/src/main/java/be/sportgreenmoove/app/ui/PaymentsScreen.kt
package be.sportgreenmoove.app.ui

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import be.sportgreenmoove.app.data.PayableBookingSummary
import be.sportgreenmoove.app.design.Sgm
import be.sportgreenmoove.app.design.SgmColor
import be.sportgreenmoove.app.design.SgmGridTexture
import be.sportgreenmoove.app.design.SgmRadius
import be.sportgreenmoove.app.design.SgmType

@Composable
fun PaymentsScreen(
    bookings: List<PayableBookingSummary>,
    loading: Boolean,
    onBack: () -> Unit,
    onPay: (PayableBookingSummary) -> Unit,
) {
    V2Screen {
        V2TopBar("PAIEMENTS", onBack = onBack)
        PaymentsHero(bookings = bookings)
        V2SectionLabel("À RÉGLER")
        if (bookings.isEmpty()) {
            PaymentsEmptyCard()
        } else {
            Column(
                modifier = Modifier.padding(horizontal = 20.dp),
                verticalArrangement = Arrangement.spacedBy(10.dp),
            ) {
                bookings.forEach { booking ->
                    PaymentBookingCard(
                        booking = booking,
                        loading = loading,
                        onPay = { if (!loading) onPay(booking) },
                    )
                }
            }
        }
        Box(Modifier.padding(bottom = 76.dp))
    }
}

@Composable
private fun PaymentsHero(bookings: List<PayableBookingSummary>) {
    val totalCents = bookings.sumOf { it.amountCents }
    Box(
        modifier = Modifier
            .padding(start = 20.dp, top = 4.dp, end = 20.dp, bottom = 16.dp)
            .fillMaxWidth()
            .clip(RoundedCornerShape(20.dp))
            .background(SgmColor.HeroGradient)
            .padding(20.dp),
    ) {
        SgmGridTexture()
        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            Text("STRIPE PAYMENTSHEET", style = SgmType.Label.copy(color = SgmColor.TextOnGreen.copy(alpha = 0.52f), fontSize = 11.sp, letterSpacing = 0.14.em))
            Text(formatEuros(totalCents), style = SgmType.Display3XL.copy(color = SgmColor.Green, fontSize = 42.sp))
            Text(
                "${bookings.size} réservation${if (bookings.size > 1) "s" else ""} approuvée${if (bookings.size > 1) "s" else ""} en attente de paiement.",
                style = SgmType.BodySM.copy(color = SgmColor.TextOnGreen.copy(alpha = 0.68f), fontSize = 13.sp, fontWeight = FontWeight.SemiBold),
            )
        }
    }
}

@Composable
private fun PaymentBookingCard(
    booking: PayableBookingSummary,
    loading: Boolean,
    onPay: () -> Unit,
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(SgmRadius.LG))
            .background(Sgm.colors.bgCard)
            .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(SgmRadius.LG))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(12.dp)) {
            V2Avatar(initials = booking.club.take(2), size = 42)
            Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(3.dp)) {
                Text(booking.title, style = SgmType.BodyBase.copy(color = Sgm.colors.textPrimary, fontSize = 15.sp, fontWeight = FontWeight.SemiBold), maxLines = 1, overflow = TextOverflow.Ellipsis)
                Text("${booking.dateLabel} · ${booking.timeLabel} · ${booking.seats} place${if (booking.seats > 1) "s" else ""}", style = SgmType.BodyXS.copy(color = Sgm.colors.textMuted, fontSize = 11.sp, fontWeight = FontWeight.Medium))
            }
            Text(booking.amountLabel, style = SgmType.DisplayXL.copy(color = SgmColor.Orange, fontSize = 22.sp))
        }
        V2Button(
            title = if (loading) "Préparation..." else "Payer maintenant",
            onClick = onPay,
            variant = V2ButtonVariant.Primary,
            size = V2ButtonSize.Md,
            full = true,
        )
    }
}

@Composable
private fun PaymentsEmptyCard() {
    Row(
        modifier = Modifier
            .padding(horizontal = 20.dp)
            .fillMaxWidth()
            .clip(RoundedCornerShape(SgmRadius.LG))
            .background(Sgm.colors.bgCard)
            .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(SgmRadius.LG))
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Box(
            modifier = Modifier
                .size(42.dp)
                .clip(RoundedCornerShape(14.dp))
                .background(SgmColor.Green.copy(alpha = 0.12f)),
            contentAlignment = Alignment.Center,
        ) {
            SgmLineIcon(SgmIcon.Award, tint = SgmColor.Green, modifier = Modifier.size(20.dp))
        }
        Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
            Text("Aucun paiement dû", style = SgmType.BodyBase.copy(color = Sgm.colors.textPrimary, fontSize = 15.sp, fontWeight = FontWeight.SemiBold))
            Text("Les réservations apparaissent ici après validation du conducteur.", style = SgmType.BodyXS.copy(color = Sgm.colors.textMuted, fontSize = 12.sp, fontWeight = FontWeight.Medium))
        }
    }
}

private fun formatEuros(cents: Int): String =
    "%.2f€".format(cents.toDouble() / 100.0).replace(".", ",")

File: ios/Sources/SportsGreenMooveApp/PaymentsScreen.swift
import SwiftUI

struct PaymentsScreen: View {
    @Environment(AppState.self) private var appState
    @Environment(\.openURL) private var openURL

    var body: some View {
        SGMScreen(testID: UITestIdentifier.paymentsScreen) {
            SGMTopBar(title: "PAIEMENTS", showsBack: true)
            PaymentsHero(bookings: appState.payableBookings)
            if appState.selectedRole == .driver {
                SGMSectionLabel("STRIPE CONNECT")
                PaymentsConnectCard(loading: appState.loading) {
                    Task { await startStripeConnect() }
                }
                .padding(.horizontal, SGMSpace.padScreen)
            }
            SGMSectionLabel("À RÉGLER")
            if appState.payableBookings.isEmpty {
                PaymentsEmptyCard()
                    .padding(.horizontal, SGMSpace.padScreen)
            } else {
                VStack(spacing: 10) {
                    ForEach(appState.payableBookings) { booking in
                        PaymentBookingCard(
                            booking: booking,
                            loading: appState.loading
                        ) {
                            Task { await pay(booking) }
                        }
                    }
                }
                .padding(.horizontal, SGMSpace.padScreen)
            }
        }
        .task {
            await appState.refreshAppData()
        }
    }

    private func startStripeConnect() async {
        appState.loading = true
        appState.errorMessage = nil
        defer { appState.loading = false }

        do {
            guard appState.stripe.isConfigured else {
                throw ProviderConfigurationError(message: "Stripe iOS n'est pas configuré.")
            }
            guard let email = appState.session?.email, !email.isEmpty else {
                throw ProviderConfigurationError(message: "Adresse email Firebase requise pour Stripe Connect.")
            }
            guard let urls = NativeStripeConnectConfiguration.urls else {
                throw ProviderConfigurationError(message: "Configurez SGM_STRIPE_CONNECT_RETURN_URL et SGM_STRIPE_CONNECT_REFRESH_URL.")
            }

            let account = try await appState.stripe.createStripeAccount(email: email)
            let link = try await appState.stripe.createStripeAccountLink(
                returnUrl: urls.returnUrl,
                refreshUrl: urls.refreshUrl
            )
            guard let url = URL(string: link.url) else {
                throw ProviderConfigurationError(message: "Lien onboarding Stripe invalide.")
            }
            openURL(url)
            appState.noticeMessage = account.reused ? "Onboarding Stripe repris." : "Compte Stripe créé."
        } catch {
            appState.errorMessage = error.localizedDescription
        }
    }

    private func pay(_ booking: PayableBookingSummary) async {
        appState.loading = true
        appState.errorMessage = nil
        defer { appState.loading = false }

        do {
            guard appState.stripe.isConfigured else {
                throw ProviderConfigurationError(message: "Stripe iOS n'est pas configuré.")
            }
            let config = try await appState.stripe.prepareRidePayment(bookingId: booking.id)
            let outcome = try await NativePaymentSheetFlow.present(config: config)
            switch outcome {
            case .completed:
                appState.noticeMessage = "Paiement confirmé."
                await appState.refreshAppData()
            case .canceled:
                appState.noticeMessage = "Paiement annulé."
            }
        } catch {
            appState.errorMessage = error.localizedDescription
        }
    }
}

private struct PaymentsConnectCard: View {
    let loading: Bool
    let action: () -> Void

    var body: some View {
        HStack(spacing: 12) {
            SGMIconView(icon: .award, size: 20, color: SGM.orange)
                .frame(width: 42, height: 42)
                .background(SGM.orange.opacity(0.12), in: RoundedRectangle(cornerRadius: 14, style: .continuous))
            VStack(alignment: .leading, spacing: 6) {
                Text("Stripe Connect")
                    .font(.sgmBody(15, weight: .semibold))
                    .foregroundStyle(SGM.textPrimary)
                Text("Activez les paiements conducteur et les futurs virements récompenses.")
                    .font(.sgmBody(12, weight: .medium))
                    .foregroundStyle(SGM.textMuted)
                SGMButton(title: loading ? "Ouverture..." : "Continuer l'onboarding", action: action)
                    .opacity(loading ? 0.58 : 1)
                    .allowsHitTesting(!loading)
            }
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(SGM.bgCard, in: RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous).stroke(SGM.border, lineWidth: 1))
    }
}

private struct PaymentsHero: View {
    let bookings: [PayableBookingSummary]

    private var totalCents: Int {
        bookings.reduce(0) { $0 + $1.amountCents }
    }

    var body: some View {
        ZStack(alignment: .leading) {
            SGM.heroGradient
            SGMGridTexture()
            VStack(alignment: .leading, spacing: 8) {
                Text("STRIPE PAYMENTSHEET")
                    .font(.sgmBody(11, weight: .bold))
                    .tracking(.sgmWider(for: 11))
                    .foregroundStyle(SGM.textOnGreen.opacity(0.52))
                Text(formatEuros(totalCents))
                    .font(.sgmDisplay(42))
                    .tracking(.sgmWide(for: 42))
                    .foregroundStyle(SGM.green)
                Text("\(bookings.count) réservation\(bookings.count > 1 ? "s" : "") approuvée\(bookings.count > 1 ? "s" : "") en attente de paiement.")
                    .font(.sgmBody(13, weight: .semibold))
                    .foregroundStyle(SGM.textOnGreen.opacity(0.68))
            }
            .padding(20)
        }
        .clipShape(RoundedRectangle(cornerRadius: 20, style: .continuous))
        .padding(.horizontal, SGMSpace.padScreen)
    }
}

private struct PaymentBookingCard: View {
    let booking: PayableBookingSummary
    let loading: Bool
    let action: () -> Void

    var body: some View {
        VStack(spacing: 12) {
            HStack(spacing: 12) {
                SGMAvatar(initials: String(booking.club.prefix(2)), size: 42)
                VStack(alignment: .leading, spacing: 3) {
                    Text(booking.title)
                        .font(.sgmBody(15, weight: .semibold))
                        .foregroundStyle(SGM.textPrimary)
                        .lineLimit(1)
                    Text("\(booking.dateLabel) · \(booking.timeLabel) · \(booking.seats) place\(booking.seats > 1 ? "s" : "")")
                        .font(.sgmBody(11, weight: .medium))
                        .foregroundStyle(SGM.textMuted)
                }
                Spacer()
                Text(booking.amountLabel)
                    .font(.sgmDisplay(22))
                    .foregroundStyle(SGM.orange)
            }
            SGMButton(title: loading ? "Préparation..." : "Payer maintenant", testID: UITestIdentifier.paymentAction, action: action)
                .opacity(loading ? 0.58 : 1)
                .allowsHitTesting(!loading)
        }
        .padding(16)
        .background(SGM.bgCard, in: RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous).stroke(SGM.border, lineWidth: 1))
    }
}

private struct PaymentsEmptyCard: View {
    var body: some View {
        HStack(spacing: 12) {
            SGMIconView(icon: .award, size: 20, color: SGM.green)
                .frame(width: 42, height: 42)
                .background(SGM.green.opacity(0.12), in: RoundedRectangle(cornerRadius: 14, style: .continuous))
            VStack(alignment: .leading, spacing: 3) {
                Text("Aucun paiement dû")
                    .font(.sgmBody(15, weight: .semibold))
                    .foregroundStyle(SGM.textPrimary)
                Text("Les réservations apparaissent ici après validation du conducteur.")
                    .font(.sgmBody(12, weight: .medium))
                    .foregroundStyle(SGM.textMuted)
            }
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(SGM.bgCard, in: RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous).stroke(SGM.border, lineWidth: 1))
    }
}

private func formatEuros(_ cents: Int) -> String {
    String(format: "%.2f€", Double(cents) / 100).replacingOccurrences(of: ".", with: ",")
}
